src/ServiceControl.Audit/Infrastructure/CheckRavenDBIndexLag.cs

```diff
@@ -0,0 +1,90 @@
+
+            var indexCountWithTooMuchLag = CheckAndReportIndexesWithTooMuchIndexLag(statistics, indexes);
+
+            if (indexCountWithTooMuchLag > 0)
+            {
+                return CheckResult.Failed($"At least one index significantly stale. Please run maintenance mode if this custom check persists to ensure index(es) can recover. See log file in `{LogPath}` for more details. Visit https://docs.particular.net/search?q=servicecontrol+troubleshooting for more information.");
+            }
+
+            return CheckResult.Pass;
+        }
+
+        static int CheckAndReportIndexesWithTooMuchIndexLag(DatabaseStatistics statistics, IndexStats[] indexes)
+        {
+            int indexCountWithTooMuchLag = 0;
+
+            foreach (var indexStats in indexes)
+            {
+                // IndexingLag is the number of documents that the index is behind, it is not a time unit.
+                var indexLag = indexStats.IndexingLag.GetValueOrDefault();
+                indexLag = Math.Abs(indexLag);
+
+                if (indexLag > IndexLagThresholdError)
+                {
+                    indexCountWithTooMuchLag++;
+                    _log.Error($"Index [{indexStats.Name}] IndexingLag {indexLag:n0} is above {IndexLagThresholdError:n0}, consider starting the instance in maintenance mode to ensure index can recover.");
+                }
+                else if (indexLag > IndexLagThresholdWarning)
+                {
+                    indexCountWithTooMuchLag++;
+                    _log.Warn($"Index [{indexStats.Name}] IndexingLag {indexLag:n0} is above {IndexLagThresholdWarning:n0}, please start the instance in maintenance mode to ensure index can recover.");
+                }
+            }
+
+            foreach (var indexError in statistics.Errors)
+            {
+                _log.Error($"Index [{indexError.IndexName}] Error: {indexError.Error} (Action: {indexError.Action},  Doc: {indexError.Document}, At: {indexError.Timestamp})");
+            }
+
+            return indexCountWithTooMuchLag;
+        }
+
+        static void CreateDiagnosticsLogEntry(DatabaseStatistics statistics, IndexStats[] indexes)
+        {
+            var report = new StringBuilder();
+            report.AppendLine("Internal RavenDB index health report:");
+
+            foreach (var indexStats in indexes)
+            {
+                // IndexingLag is the number of documents that the index is behind, it is not a time unit.
+                var indexLag = indexStats.IndexingLag.GetValueOrDefault();
+                indexLag = Math.Abs(indexLag);
+                report.AppendLine($"- Index [{indexStats.Name,-44}] Stale: {statistics.StaleIndexes.Contains(indexStats.Name),-5}, Lag: {indexLag,9:n0}, Valid: {indexStats.IsInvalidIndex,-5}, LastIndexed: {indexStats.LastIndexedTimestamp:u}, LastIndexing: {indexStats.LastIndexingTime:u}");
+            }
+            _log.Debug(report.ToString());
```

Comment:
Given that this logs to debug, should we terminate early if debug logging is not enabled?